### ISSUE TITLE
TagLib cannot open WAV files if more than 7 bytes are appended

### ISSUE THREAD

**Issue by tw-git** [taglib/taglib#973]
TagLib will still open and analyze WAV files succesfully, if up to 7 bytes are appended:
```
$ echo -n "1234567" >> my1.wav
```
If you append at least 8 bytes:
```
$ echo -n "12345678" >> my2.wav
```
TagLib returns `null` when trying to open it. It should either "throw" on the first appended byte or not at all. I guess you try to read another subchunk if there are at least 8 bytes left. I suggest to not "throw" at all as appended bytes are not illegal. I would just not read over the end of the WAVE chunk, whose size is defined at offset 4 (uint32_t).

**Comment by ufleisch** [comment]
Since such WAV files with appended garbage are accepted by the "reference operating system" for WAV files (aka. Windows), I would agree that TagLib should accept such files. I have tested that editing the tags of such files is working correctly, the garbage is not affected (i.e. it remains appended), when the tags are then set to their original values, the edited file has the same contents as the original file.

**Comment by ufleisch** [comment]
Fixed with #978.
